custom_components/envisalink_new/alarm_control_panel.py

```diff
@@ -62,6 +57,7 @@ async def async_setup_entry(
     """Set up the alarm panel based on a config entry."""
     controller = hass.data[DOMAIN][entry.entry_id]
     code = entry.data.get(CONF_CODE)
+    code_arm_required = entry.data.get(CONF_CODE_ARM_REQUIRED)
```

Comment:
The setting won't exist initially and this will return None.  So you'll need to check for that and use the current behaviour (set this based on whether they have a code configured) as the default.  This will ensure people upgrading to this version won't see a behaviour change and won't be required to change anything unless they want to take advantage of the changes.

Reply:
I don't know how to test this as I don't have a working setup that was created before these changes, but should be good now.

Reply:
Resolving as the configurable code_arm_require_arm idea has been waived.